src/applications/mhv-medical-records/containers/DownloadReportPage.jsx

```diff
@@ -140,33 +141,44 @@ const DownloadReportPage = ({ runningUnitTest }) => {
 
   const generateSEIPdf = useCallback(
     async () => {
-      setSelfEnteredPdfRequested(true);
-      setSeiPdfGenerationError(false);
+      try {
+        setSelfEnteredPdfRequested(true);
+        setSeiPdfGenerationError(false);
 
-      if (!isDataFetched) {
-        // Fetch data if not all defined
-        dispatch(clearFailedList());
-        dispatch(getSelfEnteredData());
-      } else {
-        // If already defined, generate the PDF directly
-        setSelfEnteredPdfRequested(false);
-        const title = 'Self-entered information report';
-        const subject = 'VA Medical Record';
-        const scaffold = generatePdfScaffold(userProfile, title, subject);
-        const pdfName = `VA-self-entered-information-report-${getNameDateAndTime(
-          userProfile,
-        )}`;
+        if (!isDataFetched) {
+          // Fetch data if not all defined
+          dispatch(clearFailedList());
+          dispatch(getSelfEnteredData());
+        } else {
+          // If already defined, generate the PDF directly
+          setSelfEnteredPdfRequested(false);
+          const title = 'Self-entered information report';
+          const subject = 'VA Medical Record';
+          const scaffold = generatePdfScaffold(userProfile, title, subject);
+          const pdfName = `VA-self-entered-information-report-${getNameDateAndTime(
+            userProfile,
+          )}`;
 
-        const pdfData = {
-          recordSets: generateSelfEnteredData(seiRecords),
-          ...scaffold,
-          name,
-          dob,
-          lastUpdated: UNKNOWN,
-        };
-        makePdf(pdfName, pdfData, title, runningUnitTest, 'selfEnteredInfo')
-          .then(() => setSuccessfulSeiDownload(true))
-          .catch(() => setSeiPdfGenerationError(true));
+          Object.keys(seiRecords).forEach(key => {
+            const item = seiRecords[key];
+            if (item && Array.isArray(item) && !item.length) {
+              seiRecords[key] = null;
+            }
+          });
+
+          const pdfData = {
+            recordSets: generateSelfEnteredData(seiRecords),
+            ...scaffold,
+            name,
+            dob,
+            lastUpdated: UNKNOWN,
+          };
+          makePdf(pdfName, pdfData, title, runningUnitTest, 'selfEnteredInfo')
+            .then(() => setSuccessfulSeiDownload(true))
+            .catch(() => setSeiPdfGenerationError(true));
+        }
+      } catch (error) {
+        dispatch(addAlert(ALERT_TYPE_BB_ERROR, error));
       }
```

Comment:
When you run `dispatch(addAlert(ALERT_TYPE_BB_ERROR, error))` it will display the Medical Records Download error instead of a Self-Entered error

Reply:
Fixed